.github/workflows/python-release.yml

```diff
@@ -147,23 +147,23 @@ jobs:
         run: python3 -m poetry build --format=sdist
         if: startsWith(matrix.os, 'ubuntu')
 
+      - uses: astral-sh/setup-uv@v4 # required to use `uv` in cibuildwheel
+
```

Comment:
I'm a bit reluctant to only use `uv` just for building the wheels. I'd be in favor of replacing Poetry with uv first and see if that's possible. I'm hearing a lot of good stuff about uv, bit some blogs also suggest that [it is still lacking](https://www.loopwerk.io/articles/2024/python-poetry-vs-uv/) some functionality.

Reply:
backed out of the uv changes, i created #1602 so we can take a look in the future